{Release} Upgrade to Azure CLI 2.28.0 (#19441)
Azure/azure-cli: src/azure-cli/HISTORY.rst

Release History
===============

2.28.0
++++++

**ACR**

* `az acr create/update`: Add support for disabling export through `--allow-exports` (#19065)
* `az acr`: Bump core api-version to `2021-06-01-preview` from `2020-11-01-preview`. agent_pool, tasks and runs operations unchanged from `2019-06-01-preview` (#19065)
* `az acr task credential`: Fix the issue where task credentials were not used (#19241)
* `az acr task logs`: Fix the AttributeError when querying the task logs (#19366)

**ACS**

* [BREAKING CHANGE] `az aks nodepool update`: Change rejecting the ability to use max-surge with node-image-only (#19295)

**AKS**

* `az aks install-cli`: Add support for kubelogin darwin/arm64 releases (#19072)
* Fix incorrectly passed parameter for option `--assign-kubelet-identity` in aks create sub-command (#19157)
* Upgrade api-version to `2021-07-01` for ACS module (#19245)
* `az aks create/update`: Add support for private cluster public fqdn feature (#19302)

> * Revert PR #18825: az aks create/update: Add parameter --auto-upgrade-channel to support auto upgrade (with fix) (#19297)

Reviewer: ```suggestion
* Revert PR #18825: `az aks create/update`: Add parameter `--auto-upgrade-channel` to support auto upgrade (with fix) (#19297)
```

* `aks create/aks nodepool add`: Add parameter ` --os-sku` to support choosing the underlying container host OS (#19374)

> **AMS**
> 
> * Update python to track 2 (#18687)
> 

Reviewer 2: ```suggestion
```

**App Service**

* `az webapp config storage-account list/add/update/delete`: Remove preview flag (#19177)
* Fix #18497: `functionapp identity show`: Fix the crashes when the functionapp name does not reference an existing functionapp (#19127)
* `az webapp config set`: Add additional help examples for powershell users (#19185)
* Fix #17818: `az functionapp update`:  Add instance validation for updating functionapp (#19141)
* `az webapp config hostname add`: Fix the issue caused by AttributeError (#19188)
* `az webapp config hostname add`: Fix the issue caused by AttributeError (#19348)
* Fix #16470: `az staticwebapp secrets`: Add commands to manage deployment secrets (#19347)
* `az webapp deployment source config-local-git`: Fix the issue caused by AttributeError when slot option is specified (#19404)
* `az webapp deleted restore` : Fix the issue that 'WebAppsOperations' object has no attribute 'restore_from_deleted_app' (#19404)
* `az webapp up`: Add ability to deploy Linux and Windows webapps to the same resource group (#19344)
* `az webapp up`: Add support for deploying to an App Service Environment (#19425)

> * : Fix #19098: `az webapp deployment slot auto-swap `: Fix the AttributeError error for parameters `--slot --disable` (#19376)

Reviewer 3: ```suggestion
* Fix #19098: `az webapp deployment slot auto-swap `: Fix the AttributeError error for parameters `--slot --disable` (#19376)
```

Release History
===============

2.28.0
++++++

**ACR**

> * : `az acr create/update`: Add support for disabling export through `--allow-exports` (#19065)
> * : `az acr`: Bump core api-version to `2021-06-01-preview` from `2020-11-01-preview`. agent_pool, tasks and runs operations unchanged from `2019-06-01-preview` (#19065)

Reviewer 3: ```suggestion
* `az acr create/update`: Add support for disabling export through `--allow-exports` (#19065)
* `az acr`: Bump core api-version to `2021-06-01-preview` from `2020-11-01-preview`. agent_pool, tasks and runs operations unchanged from `2019-06-01-preview` (#19065)
```

* `az acr task credential`: Fix the issue where task credentials were not used (#19241)
* `az acr task logs`: Fix the AttributeError when querying the task logs (#19366)

**ACS**

* [BREAKING CHANGE] `az aks nodepool update`: Change rejecting the ability to use max-surge with node-image-only (#19295)

**AKS**

* `az aks install-cli`: Add support for kubelogin darwin/arm64 releases (#19072)
* Fix incorrectly passed parameter for option `--assign-kubelet-identity` in aks create sub-command (#19157)
* Upgrade api-version to `2021-07-01` for ACS module (#19245)
* `az aks create/update`: Add support for private cluster public fqdn feature (#19302)
* Revert PR #18825: az aks create/update: Add parameter --auto-upgrade-channel to support auto upgrade (with fix) (#19297)

> * `aks create\aks nodepool add`: Add parameter ` --os-sku` to support choosing the underlying container host OS (#19374)

Reviewer 4: ```suggestion
* `aks create/aks nodepool add`: Add parameter `--os-sku` to support choosing the underlying container host OS (#19374)
```

> * `aks create\aks nodepool add`: Add parameter ` --os-sku` to support choosing the underlying container host OS (#19374)

Reviewer 3: ```suggestion
* `aks create/aks nodepool add`: Add parameter ` --os-sku` to support choosing the underlying container host OS (#19374)
```

**AMS**

* Update python to track 2 (#18687)

**App Config**

* `appconfig kv import/export`: Add endpoint validation during import and export (#19145)

**App Service**

> * : `az webapp config storage-account list/add/update/delete`: Remove preview flag (#19177)
> * Fix #18497: `functionapp identity show`: Fix the crashes when the functionapp name does not reference an existing functionapp (#19127)
> * :  az webapp config set : Add additional help examples for powershell users (#19185)
> * : Fix #17818: `az functionapp update`:  Add instance validation for updating functionapp (#19141)
> * `az webapp config hostname add`: Fix the issue caused by AttributeError (#19188)
> * `az webapp config hostname add`: Fix the issue caused by AttributeError (#19348)
> * : Fix #16470: `az staticwebapp secrets`: Add commands to manage deployment secrets (#19347)
> * : `az webapp deployment source config-local-git`: Fix the issue caused by AttributeError when slot option is specified (#19404)
> * : `az webapp deleted restore` : Fix the issue that 'WebAppsOperations' object has no attribute 'restore_from_deleted_app' (#19404)
> * `az webapp up`: Add ability to deploy Linux and Windows webapps to the same resource group (#19344)
> * `az webapp up`: Add support for deploying to an App Service Environment (#19425)
> * : Fix #19098: `az webapp deployment slot auto-swap `: Fix the AttributeError error for parameters `--slot --disable` (#19376)

Reviewer 3: ```suggestion
* `az webapp config storage-account list/add/update/delete`: Remove preview flag (#19177)
* Fix #18497: `functionapp identity show`: Fix the crashes when the functionapp name does not reference an existing functionapp (#19127)
* `az webapp config set`: Add additional help examples for powershell users (#19185)
* Fix #17818: `az functionapp update`:  Add instance validation for updating functionapp (#19141)
* `az webapp config hostname add`: Fix the issue caused by AttributeError (#19188)
* `az webapp config hostname add`: Fix the issue caused by AttributeError (#19348)
* Fix #16470: `az staticwebapp secrets`: Add commands to manage deployment secrets (#19347)
* `az webapp deployment source config-local-git`: Fix the issue caused by AttributeError when slot option is specified (#19404)
* `az webapp deleted restore` : Fix the issue that 'WebAppsOperations' object has no attribute 'restore_from_deleted_app' (#19404)
* `az webapp up`: Add ability to deploy Linux and Windows webapps to the same resource group (#19344)
* `az webapp up`: Add support for deploying to an App Service Environment (#19425)
* : Fix #19098: `az webapp deployment slot auto-swap `: Fix the AttributeError error for parameters `--slot --disable` (#19376)
```

**ARM**

* `az feature registration`: Add az feature registration apis (#19146)
* `az tag create`: Add the note for handling existing tag in help (#19057)
* `az ts create`: Fix issue where creating a template spec with inner deployments that reference a common template fails (#19312)

**CDN**

* `az cdn endpoint create`: Fix endpoint creation failure with `--content-types-to-compress` (#19401)

> **Compute **

Reviewer: ```suggestion
**Compute**
```

* `az ssh vm`: Raise error for managed identity and Cloud Shell (#19051)
* Upgrade api-version for VM and VMSS from `2021-03-01` to `2021-04-01` (#19158)
* `az vmss create/update`: Support spot restore policy to VM scale sets (#19189)
* Add new examples for creating disk from share image gallery (#19270)

> * `az vm image​ list\list-offers\list-skus\list-publishers\show`: Add new parameter ​`--edge-zone`​ to support querying the image under edge zone (#19206)

Reviewer: ```suggestion
* `az vm image​ list/list-offers/list-skus/list-publishers/show`: Add new parameter ​`--edge-zone`​ to support querying the image under edge zone (#19206)
```

* `az capacity reservation group`: Add new commands to manage capacity reservation group (#19416)
* `az vm create/update`: Add new parameter `--capacity-reservation-group` to support association to capacity reservation (#19416)
* `az vmss create/update`: Add new parameter `--capacity-reservation-group` to support association to capacity reservation (#19416)
* `az vmss create`: Support creating VMSS from shared gallery image (#19417)

**IoT**

* `az iot hub/dps certificate update/create`: Add `--verified` argument to mark certificates as verified without proof-of-possession flow (#19363)
* `az iot hub create/update`: Add `--disable-local-auth`, `--disable-device-sas`, and `--disable-module-sas` arguments to configure accepted SAS key authentication methods. (#19363)

**Key Vault**

* `az keyvault private-endpoint-connection list`: Support list mhsm's private endpoint connections (#19163)
* `az keyvault set-policy`: `--key-permissions` add new option `release` (#19411)

**Network**

* Fix NSG rule creation example mistake (#19242)
* Add a new command group `az network custom-ip prefix`. (#19218)
* `az network public-ip`: Add parameter `--ip-address`. (#19218)

> * `network public-ip prefix create`: Add parameter `--custom-ip-prefix-name`. (#19218)

Reviewer 3: ```suggestion
* `az network public-ip prefix create`: Add parameter `--custom-ip-prefix-name`. (#19218)
```

> * Fix'az network dns record-set [record-type] add-record' is not idempotent (#19237)

Reviewer 5: ```suggestion
* Fix `az network dns record-set [record-type] add-record` is not idempotent (#19237)
```

* Drop Ubuntu 20.10 Groovy Gorilla support (#19368)

**PowerBI**

* Add private link provider Microsoft.PowerBI/privateLinkServicesForPowerBI (#19204)

**RDBMS**

* [BREAKING CHANGE] `az postgres flexible-server migration`: Rename `--migration-id` to `--migration-name` (#19149)
* [BREAKING CHANGE] `az mysql flexible-server create/update`: `--high-availability` available parameter is changed from 'Enabled' to 'ZoneRedundant' and 'SameZone' . (#19301)
* Fix maintenance window update issue with MySQL and Change restart parameter to be case insensitive (#19231)
* `az mysql flexible-server restore` enables network option change from private network  to public network and vice versa. (#19301)
* `az mysql flexible-server replica create`: Add `zone` parameter. (#19301)

**Role**

* `az role assignment create`: Support `ForeignGroup` for `--assignee-principal-type` (#19132)
* `az role assignment create`: Do not invoke Graph API if `--assignee-principal-type` is provided (#19219)

**SQL**

* `az sql mi update`: Add --subnet and --vnet-name parameters to support the cross subnet update SLO (#18886)
* Fix the enum name change in track2 Python SDK (#19142)

**Storage**

> * Fix #10765 to refine error message when account key is incorrect padding (#13965)

Reviewer 6: ```suggestion
* Fix #10765: Refine error message when account key is incorrect padding (#13965)
```

**Synapse**

> * [BREAKING CHANGE] az synapse workspace key activate: Make command more clearer (#19304)

Reviewer 5: ```suggestion
* [BREAKING CHANGE] `az synapse workspace key activate`: Make command more clearer (#19304)
```

> * Code owner change (#19119)

Reviewer 6: ```suggestion
* [BREAKING CHANGE] `az synapse workspace key activate`: Make command clearer (#19304)
```

> * [BREAKING CHANGE] `az synapse workspace key activate`: Make command more clearer (#19304)
> * Code owner change (#19119)

Reviewer 6: ```suggestion
* [BREAKING CHANGE] Rename `az synapse workspace key update` to `az synapse workspace key activate` and remove `--is-active` (#19304)
```
